File: data/gpt-4o/readable_format_HPEC/cpp/file191.cpp
void init_array(int ni, int nj, double a[][100], double r[][100], double q[][100])
{
 int i, j;
 for (i = 0; i < ni; i++)
 for (j = 0; j < nj; j++)
 {
 a[j][i] = ((double)(i) * (double)(j)) / (double)ni;
 q[j][i] = ((double)(i) * (double)(j+1)) / (double)nj;
 }
 for (i = 0; i < ni; i++)
 for (j = 0; j < nj; j++)
 r[j][i] = ((double)(i) * (double)(j+2)) / (double)nj;
}

File: data/StarCoder2/readable_format_HPEC/fortran/file191.f90
subroutine init_array(ni, nj, a, r, q)
implicit none

DATA_TYPE, dimension(nj, ni) :: a
DATA_TYPE, dimension(nj, nj) :: r
DATA_TYPE, dimension(nj, ni) :: q
integer :: ni, nj
integer :: i, j

do i = 1, ni
do j = 1, nj
a(j, i) = (DBLE(i - 1) * DBLE(j - 1)) / DBLE(ni)
q(j, i) = (DBLE(i - 1) * DBLE(j)) / DBLE(nj)
end do
end do

do i = 1, ni
do j = 1, nj
r(j, i) = (DBLE(i - 1) * DBLE(j + 1)) / DBLE(nj)
end do
end do
end subroutine
